big-o-notation/big-o.js

```diff
@@ -0,0 +1,44 @@
+/* exported uniqueQuadratic, uniqueLinear */
+
+function uniqueLinear(words) {
+  const seen = {};                  // 1 * 1 = O(1)
+  const unique = [];                // 1 * 1 = O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(2n) ~= O(n)
+    i++                             // 2 * n = O(2n) ~= O(n)
+  ) {
+    const word = words[i];          // 2 * n = O(2n) ~= O(n)
+    if (!seen[word]) {              // 2 * n = O(2n) ~= O(n)
+      seen[word] = true;
+      unique[unique.length] = word; // 2 * n = O(2n) ~= O(n)
+    }
+  }
+  return unique;
+} // Big O Notation for uniqueLinear: O(n)
+
+function uniqueQuadratic(words) {
+  const unique = [];                // 1 * 1 = O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(2) ~= O(n)
+    i++                             // 2 * n = O(2) ~= O(n)
+  ) {
+    const word = words[i];          // 2 * n = O(2n) ~= O(n)
+    let isUnique = true;            // 1 * 1 = O(1)
+    for (
+      let c = 0;                    // 1 * 1  = O(1)
+      c < i;                        // 2 * n^2 = 2(n^2) ~= (n^2)
+      c++                           // 2 * n^2 = 2(n^2) ~= (n^2)
+    ) {
+      const comparing = words[c];   // 2 * 1 = O(2)
```

Comment:
This is inside the inner loop so will be: `2 * n^2 = O(2(n^2)) ~= O(n^2)`

Reply:
Okay, that makes sense.